filebeat/harvester/processor/line.go

```diff
@@ -0,0 +1,90 @@
+package processor
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// LineSource produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type LineSource struct {
+	reader *encoding.LineReader
+}
+
+// Line represents a line event with timestamp, content and actual number
+// of bytes read from input before decoding.
+type Line struct {
+	Ts      time.Time     // timestamp the line was read
+	Content []byte        // actual line read
+	Bytes   int           // total number of bytes read to generate the line
+	Fields  common.MapStr // optional fields that can be added by processors
+}
+
+// LineProcessor is the interface that wraps the basic Next method for
+// getting a new line.
+// Next returns the line being read or and error. EOF is returned
+// if processor will not return any new lines on subsequent calls.
+type LineProcessor interface {
+	Next() (Line, error)
+}
+
+// StripNewline processor removes the last trailing newline characters from
+// read lines.
+type StripNewline struct {
```

Comment:
`StripNewLine` is a special `LineProcessor` removing newline character. Can be moved into separate go file.